services/app-web/src/pages/StateSubmission/ReviewSubmit/V2/ReviewSubmit/RateDetailsSummarySectionV2.tsx

```diff
@@ -0,0 +1,319 @@
+    submissionName: string
+    statePrograms: Program[]
+    onDocumentError?: (error: true) => void
+}
+
+export function renderDownloadButton(
+    zippedFilesURL: string | undefined | Error
+) {
+    if (zippedFilesURL instanceof Error) {
+        return (
+            <InlineDocumentWarning message="Rate document download is unavailable" />
+        )
+    }
+    return (
+        <DownloadButton
+            text="Download all rate documents"
+            zippedFilesURL={zippedFilesURL}
+        />
+    )
+}
+
+export const RateDetailsSummarySectionV2 = ({
+    contract,
+    editNavigateTo,
+    documentDateLookupTable,
+    submissionName,
+    statePrograms,
+    onDocumentError,
+}: RateDetailsSummarySectionV2Props): React.ReactElement => {
+    const isSubmitted = contract.status === 'SUBMITTED'
+    const isEditing = !isSubmitted && editNavigateTo !== undefined
+    const isPreviousSubmission = usePreviousSubmission()
+    const contractFormData =
+        contract.draftRevision?.formData ||
+        contract.packageSubmissions[0].contractRevision.formData
+    const rates =
+        contract.draftRates || contract.packageSubmissions[0].rateRevisions
+    const { getKey, getBulkDlURL } = useS3()
+    const [zippedFilesURL, setZippedFilesURL] = useState<
+        string | undefined | Error
+    >(undefined)
+
+    const rateCapitationType = (rate: Rate | RateRevision) => {
+        const rateFormData = getRateFormData(rate)
+        return rateFormData?.rateCapitationType
+            ? rateFormData.rateCapitationType === 'RATE_CELL'
+                ? 'Certification of capitation rates specific to each rate cell'
+                : 'Certification of rate ranges of capitation rates per rate cell'
+            : ''
+    }
+
+    const ratePrograms = (rate: Rate | RateRevision) => {
+        /* if we have rateProgramIDs, use them, otherwise use programIDs */
+        let programIDs = [] as string[]
+        const rateFormData = getRateFormData(rate)
+
+        if (
+            rateFormData?.rateProgramIDs &&
+            rateFormData?.rateProgramIDs.length > 0
+        ) {
+            programIDs = rateFormData.rateProgramIDs
+        } else if (
+            contractFormData?.programIDs &&
+            contractFormData?.programIDs.length > 0
+        ) {
+            programIDs = contractFormData.programIDs
+        }
+        return programIDs
+            ? statePrograms
+                  .filter((p) => programIDs.includes(p.id))
+                  .map((p) => p.name)
+            : undefined
+    }
+
+    const rateCertificationType = (rate: Rate | RateRevision) => {
+        const rateFormData = getRateFormData(rate)
+        if (rateFormData?.rateType === 'AMENDMENT') {
+            return 'Amendment to prior rate certification'
+        }
+        if (rateFormData?.rateType === 'NEW') {
+            return 'New rate certification'
+        }
+    }
+
+    const getRateFormData = (rate: Rate | RateRevision) => {
```

Comment:
I'd love for you to explicitly define these functions's return types. The fact that they can return undefined leads to a bunch of ?'s later down in the code that might not be necessary.